### ISSUE TITLE
[IOS]: Pods/MLImage/Frameworks/MLImage.framework/MLImage' for architecture arm64 

### ISSUE THREAD

**Issue by songhee24** [juliansteenbakker/mobile_scanner#585]
### when i start building ios project I get an error:

```
↳
    ** BUILD FAILED **


Xcode's output:
↳
    Writing result bundle at path:
    	/var/folders/rn/r4vcnfnn7zj7gwx5w7fbgldh0000gn/T/flutter_tools.K8VYGY/flutter_ios_build_temp_diroPUaJI/temporary_xcresult_bundle

    ld: warning: ignoring file /Users/mac/WebstormProjects/srs/build/ios/Debug-iphonesimulator/Pods_Runner.framework/Pods_Runner, building for iOS Simulator-arm64 but attempting to link with file built for iOS Simulator-x86_64
    ld: in /Users/mac/WebstormProjects/srs/ios/Pods/MLImage/Frameworks/MLImage.framework/MLImage(GMLImage.o), building for iOS Simulator, but linking in object file built for iOS, file '/Users/mac/WebstormProjects/srs/ios/Pods/MLImage/Frameworks/MLImage.framework/MLImage' for architecture arm64
    clang: error: linker command failed with exit code 1 (use -v to see invocation)
    note: Building targets in dependency order
    note: Run script build phase 'Thin Binary' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'Runner' from project 'Runner')
    note: Run script build phase 'Run Script' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'Runner' from project 'Runner')

    Result bundle written to path:
    	/var/folders/rn/r4vcnfnn7zj7gwx5w7fbgldh0000gn/T/flutter_tools.K8VYGY/flutter_ios_build_temp_diroPUaJI/temporary_xcresult_bundle


Error (Xcode): In /Users/mac/WebstormProjects/srs/ios/Pods/MLImage/Frameworks/MLImage.framework/MLImage(GMLImage.o), building for iOS Simulator, but linking in object file built for iOS, file '/Users/mac/WebstormProjects/srs/ios/Pods/MLImage/Frameworks/MLImage.framework/MLImage' for architecture arm64

Could not build the application for the simulator.
Error launching application on iPhone 14.
```

**I tried changing the Excluded Architectures to arm64** 
<img width="865" alt="Screenshot 2023-04-10 at 18 11 14" src="https://user-images.githubusercontent.com/44497623/230899264-0f4a79f3-da91-4712-b6f7-f069f6e9417d.png">

the project is being built, but I noticed that after it was built, the simulator slows down very seriously, just the usual scroll lags
how can i fix this ? what could be the problem with this ? 

but when I uploaded it to the testflight and downloaded and checked there are no lags, does it mean that something is wrong with my laptop?

**EDITED:**

i tried to create a new flutter app which just builds a list no extra loads

<img width="275" alt="Screenshot 2023-04-11 at 18 41 06" src="https://user-images.githubusercontent.com/44497623/231165260-0b381f7d-fc0d-4652-8bf7-2c021fd2117e.png">
I also tried to change the launch runner to arm64 

the same thing, it works slowly, it lags when I try to scroll, as I understand it, this is not related to the library [mobile_scanner](https://github.com/juliansteenbakker/mobile_scanner)

but why it throws error when i try to run my app without  Excluded Architectures to arm64

**Deleting pods or flutter clean doesn't change anything**

**Comment by theDeji** [comment]
I'm having this same issue, please has a fix been found ? changed mine from armv7 to arm64 and it still threw that error

**Comment by lucasvini382** [comment]
I try run my app using rosseta but my application broke too.

**Comment by lucasvini382** [comment]
Yes @songhee24 i make this

**Comment by Tienisto** [comment]
If I exclude arm64, then the app does not run at all... I guess it is because I am using an ARM based macOS

**Comment by Tienisto** [comment]
I figured out that I need to install Rosetta 2.

**Comment by themon303** [comment]
This is [Known issues for GoogleMlkit](https://developers.google.com/ml-kit/known-issues#ios_issues).

**Comment by mrRedSun** [comment]
XCode 15 + Sonoma + latest flutter (3.13.6) + default excluded architectures in xcode = simulator working for me on M1 mac

**Comment by rdesimone** [comment]
@mrRedSun would you mind to share a screenshot of the mentioned xCode Build Settings? Thanks!

**Comment by mrRedSun** [comment]
@rdesimone sure thing
<img width="776" alt="image" src="https://github.com/juliansteenbakker/mobile_scanner/assets/22750848/156f91e0-2e26-4607-8aff-6367fa042e64">

**Comment by rdesimone** [comment]
Thanks a lot, I removed `i386` and now it works!

**Comment by mrRedSun** [comment]
So the workaround is to remove i386 and the issue disappears. No sure if we can close the ticket yet, but nice to know

**Comment by rdesimone** [comment]
it was the only difference I could see between yours and my settings.

**Comment by Hilbert2048** [comment]
I have the same issue, when introducing mobile scanner pub, my App is very slow in iOS simulator...

**Comment by AnthonyWharton** [comment]
Hi all,

I am working on a completely separate project with XCode but have found a workaround for getting this working on simulators on Apple Silicon Macs. You need to run the project built for x86_64 for simulator builds, and then run the simulator in Rosetta. Until Google fix the known issue linked by @themon303 there is no getting around this.

1) Set up the architecture for your project (not the individual target!), **and** the Pods project (not a target!) to replicate the following:

    ![image](https://github.com/juliansteenbakker/mobile_scanner/assets/7091722/d4f39e37-f576-4634-a67f-e01765076713)

    **Of importance**: `Any iOS Simulator SDK` should be set to x86_64 and on debug you should enable `Build Active Architecture Only`. Also note I am not excluding any architectures, [Apple themselves do not recommend doing this unless absolutely necessary](https://developer.apple.com/documentation/technotes/tn3117-resolving-build-errors-for-apple-silicon).

2) Check that the Pods config file is not intefering and adding an excluded architecture by lookind under `Pods` -> `Targets Support Files` -> `Pods-[PROJECTNAME]` -> `Pods-[PROJECTNAME].debug.xcconfig`

    ![image](https://github.com/juliansteenbakker/mobile_scanner/assets/7091722/91bc11fa-5fe3-4186-9b46-17c77338b6bd)

3) Unhide Rosetta by going to `Product` -> `Destination` -> `Destination Architectures` -> `Show Both`

    ![image](https://github.com/juliansteenbakker/mobile_scanner/assets/7091722/f0cc0578-ffd6-4aa3-b486-ebe9368a49f0)

4) Select a Rosetta Simulator (e.g. `iPhone 15 Pro (Rosetta)`), and run! (might be advisable to clean the build folder prior to running but it should work regardless).

Good luck!

**Comment by kyle-seongwoo-jun** [comment]
FYI, here is the open issue link of Google issue: [issuetracker.google.com/issues/17896515](https://issuetracker.google.com/issues/178965151)

**Comment by hpelitebook745G2** [comment]
thanks !@AnthonyWharton! As of Xcode 15.4, `Destination Architecture` doesn't exist anymore in step 3. It's now:
```
Product -> Destination -> Show All Run Destinations
```

**Comment by AnthonyWharton** [comment]
Thanks @hpelitebook745G2, I've updated the steps and added a note about the screenshot being out of date!

**Comment by RutgerVleuten** [comment]
Would be better to have an alternative for MLImage.framework, any options here?

**Comment by antoinepemeja** [comment]
Hi, I'm not going to add much to the debate. 
I can confirm the problem: on my Mac M2, if I launch my application without `mobile_scanner: ^6.0.2` it's very smooth in the iOS simulator (iPhone 16 Plus), if I add the library without even using it in my code, my whole application lags.

```
[✓] Flutter (Channel stable, 3.24.3, on macOS 14.5 23F79 darwin-arm64, locale fr-CA)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 16.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2024.1)
[✓] VS Code (version 1.94.2)
[✓] Connected device (6 available)
[✓] Network resources
```

**Comment by AroldoGoulart** [comment]
> @rdesimone sure thing <img alt="image" width="776" src="https://private-user-images.githubusercontent.com/22750848/274572555-156f91e0-2e26-4607-8aff-6367fa042e64.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk3OTU3NzIsIm5iZiI6MTcyOTc5NTQ3MiwicGF0aCI6Ii8yMjc1MDg0OC8yNzQ1NzI1NTUtMTU2ZjkxZTAtMmUyNi00NjA3LThhZmYtNjM2N2ZhMDQyZTY0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI0VDE4NDQzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQyZjI1OTExOTg4N2UwN2M1ZGQyMWIwY2FjOGMyZWM4NDhlMTRiYTMyNTBhOTRjMzZjMmQ0YTE4Zjc1OGYxMGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.7HroKv6vAczt6Xziz2mcZUKm_NJo5Mwicp12YvOC1ys">

Lol same thing work for React-native o/ thanks

**Comment by antoinepemeja** [comment]
> Hi, I'm not going to add much to the debate. I can confirm the problem: on my Mac M2, if I launch my application without `mobile_scanner: ^6.0.2` it's very smooth in the iOS simulator (iPhone 16 Plus), if I add the library without even using it in my code, my whole application lags.
> 
> ```
> [✓] Flutter (Channel stable, 3.24.3, on macOS 14.5 23F79 darwin-arm64, locale fr-CA)
> [✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
> [✓] Xcode - develop for iOS and macOS (Xcode 16.0)
> [✓] Chrome - develop for the web
> [✓] Android Studio (version 2024.1)
> [✓] VS Code (version 1.94.2)
> [✓] Connected device (6 available)
> [✓] Network resources
> ```

I tested mobile scanner version `v7.0.0-beta.3`, it is very smooth now, thank you team!

**Comment by navaronbracke** [comment]
We excluded arm64 Simulators in version 6.0.2, since MLKIt doesn't support those. Support for arm64 Simulators is available again from mobile_scanner version 7.0.0 onwards, as we dropped the dependency on MLKit.
